### DISCUSSION TITLE
Possible to stop floating windows being dragged outside visible dockview area?

### DISCUSSION THREAD

**Discussion by lexwebb** [mathuo/dockview#325] [General]
I'm new to dockview and I've been doing some experimenting in my app. I want to use a mix of floating and docked windows and I've noticed it's possible to drag the floating windows outside of the general dockview area.

Is there a built in way to prevent this behavior, and constrain the dragging of windows to only the Dockview area? I haven't been able to find anything in the docs or issues about it, but i'm happy to be proven otherwise!

**Comment by mathuo** [comment]
There is a hardcoded limit as to how far you can drag a panel outside of the main viewport which is currently set to `100px` in both height and width meaning that a minimum of `100px` in both the height and width direction must be within the main viewport. For example as seen below.

![image](https://github.com/mathuo/dockview/assets/6710312/0a42d23d-78ce-423c-8d83-e860942732f8)

> For those interested [this](https://github.com/mathuo/dockview/blob/master/packages/dockview-core/src/dockview/dockviewComponent.ts#L355) is code.

If you was to set the `overflow: hidden` on the class `dv-dockview` this would prevent the floating grouped "leaving" the viewport.

```css
.dv-dockview {
    overlay: hidden;
}
```

Adding this custom `css` would result in something like below for the same original example.

![image](https://github.com/mathuo/dockview/assets/6710312/770480c3-3db0-43ab-9869-1d121e195199)

Can I ask are you trying to simply constrain the floating groups within the viewport as with the above css example or are you trying to achieve a stricter approach where the entire floating group should remain entirely within the viewport? Either way I'm happy to include more information on this in the documentation and/or look at adding customization options for the above if required.

**Reply by lexwebb** [reply]
The latter, stricter approach. It would be nice for my use-case if the groups could be fully constrained in the bounds of the viewport. At least the header of the group, so you could move the body of the group bellow of screen but the entire header would always be visible.

**Comment by mathuo** [comment]
Addressed in `v.1.8.3` and described in the documentation [here](https://dockview.dev/docs/next/components/dockview#floating-groups)
